apps/ex_wire/lib/ex_wire/config.ex

```diff
@@ -115,6 +115,11 @@ defmodule ExWire.Config do
     get_env!(:node_discovery)
   end
 
+  @spec get_environment() :: atom() | ArgumentError
+  def get_environment() do
```

Comment:
Why don't we just use `Mix.env()` here, since the configuration looks to be identical and it's less surprising. I assume you're preparing for the issue of having multiple "environments" under prod, such as rinkeby, etc? But I'm not sure if having a `Mix.env()` and `Config.get_environment()` would necessarily would even in that case. I think we're best sticking with `Mix.env()` here and adding `Config.get_environment()` when we have a stronger case to separate it out. Thoughts?

Reply:
The problem with the Mix.* namespace is that it's not available in a OTP release. So if you have environment based conditionals, I don't know of other ways of doing this.

Reply:
So isn't it okay to use then if it gets baked in at compile that? Generating a production release will mean that it's always set to prod, which is what we would expect?

Reply:
It gets evaluated only for .exs configuration files but not inside modules. 
You can also make dev releases that are quicker or custom (like staging).